Sming/Arch/Host/Components/esp_hal/tasks.cpp

```diff
@@ -80,7 +85,13 @@ bool system_os_post(uint8_t prio, os_signal_t sig, os_param_t par)
 		return false;
 	}
 
-	return task_queues[prio]->post(sig, par);
+	if(!task_queues[prio]->post(sig, par)) {
+		return false;
+	}
+
+	extern CSemaphore host_main_loop_semaphore;
```

Comment:
@mikee47  Can you move this line to the beginning of the file?

Reply:
I'll need to rewrite this using functions anyway.
As you know from CI the Windows run keeps stalling. It happens because the Windows pthreads library uses a mutex to guard semaphores. If that mutex is locked when the main thread is suspended, the next call to sem_post() results in a deadlock. Oh, the joys of multithreading.